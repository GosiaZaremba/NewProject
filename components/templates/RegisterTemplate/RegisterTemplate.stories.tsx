/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { RegisterTemplate } from './RegisterTemplate';

storiesOf('templates/RegisterTemplate', module)
    .addDecorator((story) => story())
    .add('default', () => (
        <RegisterTemplate
            onSubmitForm={() => {}}
            onPressGoogle={() => {}}
            onPressFacebook={() => {}}
            onPressNavigateLogin={() => {}}
            onPressLinkedin={() => {}}
        />
    ));
