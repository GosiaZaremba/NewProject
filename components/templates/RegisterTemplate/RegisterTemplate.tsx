import React from 'react';
import { View } from 'react-native';
import { Label, ScreenTitle, Link, Divider } from '../../atoms';
import { SocialLogin } from '../../organisms/SocialLogin/SocialLogin';
import { RegisterForm } from '../../organisms';
import styles from './RegisterTemplate.styles';

export type Props = {
    onSubmitForm: (email: string, password: string) => void;
    onPressGoogle: () => void;
    onPressFacebook: () => void;
    onPressNavigateLogin: () => void;
    onPressLinkedin: () => void;
    testID?: string;
};

export const RegisterTemplate: React.FC<Props> = ({
    onSubmitForm,
    onPressFacebook,
    onPressGoogle,
    onPressNavigateLogin,
    onPressLinkedin,
    testID,
}) => {
    return (
        <View style={styles.mainContainer}>
            <ScreenTitle
                screenTitle={'SIGN UP'}
                testID={`${testID}-screenTitile`}
            />
            <RegisterForm onSubmitForm={onSubmitForm} testID={`${testID}`} />
            <View style={styles.divider}>
                <Divider text={'OR'} testID={`${testID}-divider`} />
            </View>
            <SocialLogin
                onPressFacebook={onPressFacebook}
                onPressGoogle={onPressGoogle}
                onPressLinkedin={onPressLinkedin}
                testID={`${testID}`}
            />
            <View style={styles.footer}>
                <Label text={'Already a user? '} testID={`${testID}-label`} />
                <Link
                    text={'LOGIN'}
                    onPress={onPressNavigateLogin}
                    testID={`${testID}-link`}
                />
            </View>
        </View>
    );
};
