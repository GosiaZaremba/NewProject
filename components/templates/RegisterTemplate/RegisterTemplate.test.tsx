import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import { RegisterTemplate } from './RegisterTemplate';

const mockOnPress = jest.fn();
const mockOnPressGoogle = jest.fn();
const mockOnPressFacebook = jest.fn();
const mockOnPressLinked = jest.fn();
const mockOnPressNavigateLogin = jest.fn();

describe('Template RegisterTemplate', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <RegisterTemplate
                    onSubmitForm={mockOnPress}
                    onPressGoogle={mockOnPressGoogle}
                    onPressFacebook={mockOnPressFacebook}
                    onPressLinkedin={mockOnPressLinked}
                    onPressNavigateLogin={mockOnPressNavigateLogin}
                    testID={'template-registerTemplate'}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure onSubmitForm works correctly', () => {
        const { getByTestId } = render(
            <RegisterTemplate
                onSubmitForm={mockOnPress}
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                onPressNavigateLogin={mockOnPressNavigateLogin}
                testID={'template-registerTemplate'}
            />
        );

        const inputEmail = getByTestId(
            'template-registerTemplate-email-input-input'
        );
        const inputPassword = getByTestId(
            'template-registerTemplate-password-input-input'
        );

        const submitButton = getByTestId(
            'template-registerTemplate-submit-button'
        );
        fireEvent.changeText(inputPassword, 'haslo');
        fireEvent.changeText(inputEmail, 'mail@wp.pl');
        fireEvent.press(submitButton);
        expect(mockOnPress).toHaveBeenCalledWith('mail@wp.pl', 'haslo');
    });

    test('ensure linkedInIcon works correctly', () => {
        const { getByTestId } = render(
            <RegisterTemplate
                onSubmitForm={mockOnPress}
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                onPressNavigateLogin={mockOnPressNavigateLogin}
                testID={'template-registerTemplate'}
            />
        );
        const linkedInIcon = getByTestId(
            'template-registerTemplate-linkedIn-icon'
        );
        expect(linkedInIcon.props.style[1].fontSize).toEqual(20);
        fireEvent.press(linkedInIcon);
        expect(mockOnPressLinked).toHaveBeenCalledTimes(1);
    });

    test('ensure googleIcon works correctly', () => {
        const { getByTestId } = render(
            <RegisterTemplate
                onSubmitForm={mockOnPress}
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                onPressNavigateLogin={mockOnPressNavigateLogin}
                testID={'template-registerTemplate'}
            />
        );

        const googleIcon = getByTestId('template-registerTemplate-google-icon');
        expect(googleIcon.props.style[1].fontSize).toEqual(20);
        fireEvent.press(googleIcon);
        expect(mockOnPressGoogle).toHaveBeenCalledTimes(1);
    });

    test('ensure facebookIcon works correctly', () => {
        const { getByTestId } = render(
            <RegisterTemplate
                onSubmitForm={mockOnPress}
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                onPressNavigateLogin={mockOnPressNavigateLogin}
                testID={'template-registerTemplate'}
            />
        );

        const facebookIcon = getByTestId(
            'template-registerTemplate-facebook-icon'
        );
        expect(facebookIcon.props.style[1].fontSize).toEqual(20);
        fireEvent.press(facebookIcon);
        expect(mockOnPressFacebook).toHaveBeenCalledTimes(1);
    });

    test('ensure screenTitle works correctly', () => {
        const { getByTestId } = render(
            <RegisterTemplate
                onSubmitForm={mockOnPress}
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                onPressNavigateLogin={mockOnPressNavigateLogin}
                testID={'template-registerTemplate'}
            />
        );

        const screenTitle = getByTestId(
            'template-registerTemplate-screenTitile'
        );
        expect(screenTitle).toHaveTextContent('SIGN UP');
    });

    test('ensure registerDivider works correctly', () => {
        const { getByTestId } = render(
            <RegisterTemplate
                onSubmitForm={mockOnPress}
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                onPressNavigateLogin={mockOnPressNavigateLogin}
                testID={'template-registerTemplate'}
            />
        );

        const registerDivider = getByTestId(
            'template-registerTemplate-divider'
        );
        expect(registerDivider).toHaveTextContent('OR');
    });

    test('ensure registerLabel works correctly', () => {
        const { getByTestId } = render(
            <RegisterTemplate
                onSubmitForm={mockOnPress}
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                onPressNavigateLogin={mockOnPressNavigateLogin}
                testID={'template-registerTemplate'}
            />
        );

        const registerLabel = getByTestId('template-registerTemplate-label');
        expect(registerLabel).toHaveTextContent('Already a user?');
    });
    test('ensure registerLink works correctly', () => {
        const { getByTestId } = render(
            <RegisterTemplate
                onSubmitForm={mockOnPress}
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                onPressNavigateLogin={mockOnPressNavigateLogin}
                testID={'template-registerTemplate'}
            />
        );

        const registerLink = getByTestId('template-registerTemplate-link');
        expect(registerLink).toHaveTextContent('LOGIN');
    });
});
