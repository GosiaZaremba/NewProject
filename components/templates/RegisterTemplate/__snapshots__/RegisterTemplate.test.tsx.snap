// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template RegisterTemplate Component Snapshot 1`] = `
<View
  style={
    Object {
      "padding": 40,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 20,
        "fontWeight": "700",
      }
    }
    testID="template-registerTemplate-screenTitile"
  >
    SIGN UP
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "marginBottom": 2,
              }
            }
            testID="template-registerTemplate-email-input-label"
          >
            Email
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="email"
            secureTextEntry={false}
            style={
              Object {
                "borderColor": "#908D8E",
                "borderRadius": 5,
                "borderWidth": 1,
                "padding": 10,
                "width": "100%",
              }
            }
            testID="template-registerTemplate-email-input-input"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "marginBottom": 2,
              }
            }
            testID="template-registerTemplate-password-input-label"
          >
            Password
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="password"
            secureTextEntry={true}
            style={
              Object {
                "borderColor": "#908D8E",
                "borderRadius": 5,
                "borderWidth": 1,
                "padding": 10,
                "width": "100%",
              }
            }
            testID="template-registerTemplate-password-input-input"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 15,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="SIGN UP"
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="template-registerTemplate-submit-button"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fc296f",
                },
              ]
            }
          >
            SIGN UP
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#908D8E",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 1,
          "justifyContent": "center",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F5F0F2",
            "borderColor": "#908D8E",
            "borderRadius": 5,
            "borderWidth": 1,
            "padding": 5,
            "position": "absolute",
          }
        }
        testID="template-registerTemplate-divider"
      >
        <Text>
          OR
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginVertical": 10,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={5}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 45,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 0.5,
            "padding": 10,
            "width": 45,
          },
          Object {
            "borderColor": "#e63039",
          },
        ]
      }
      testID="template-registerTemplate-google-pressable"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#e63039",
              "fontSize": 12,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="template-registerTemplate-google-icon"
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={5}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 45,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 0.5,
            "padding": 10,
            "width": 45,
          },
          Object {
            "borderColor": "#39539b",
          },
        ]
      }
      testID="template-registerTemplate-facebook-pressable"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#39539b",
              "fontSize": 12,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="template-registerTemplate-facebook-icon"
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={5}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 45,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 0.5,
            "padding": 10,
            "width": 45,
          },
          Object {
            "borderColor": "#3872c3",
          },
        ]
      }
      testID="template-registerTemplate-linkedIn-pressable"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#3872c3",
              "fontSize": 12,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="template-registerTemplate-linkedIn-icon"
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "marginBottom": 2,
        }
      }
      testID="template-registerTemplate-label"
    >
      Already a user? 
    </Text>
    <Text
      onPress={[MockFunction]}
      style={
        Object {
          "color": "black",
          "textDecorationLine": "underline",
        }
      }
      testID="template-registerTemplate-link"
    >
      LOGIN
    </Text>
  </View>
</View>
`;
